src/DurableTask.Netherite/StorageLayer/Faster/FasterKV.cs

```diff
@@ -950,45 +953,56 @@ public override ValueTask RemoveKeys(IEnumerable<TrackedObjectKey> keys)
         async IAsyncEnumerable<(string,OrchestrationState)> QueryEnumeratedStates(
             EffectTracker effectTracker,
             PartitionQueryEvent queryEvent,
-            IEnumerator<string> enumerator,
+            IEnumerator<string> queryRequests,
             int pageSize,
             TimeSpan timeBudget,
             DateTime attempt
            )
         {
             var instanceQuery = queryEvent.InstanceQuery;
             string queryId = queryEvent.EventIdString;
+
             int? pageLimit = pageSize > 0 ? pageSize : null;
-            this.partition.EventDetailTracer?.TraceEventProcessingDetail($"query {queryId} attempt {attempt:o} enumeration from {queryEvent.ContinuationToken} with pageLimit={(pageLimit.HasValue ? pageLimit.ToString() : "none")} timeBudget={timeBudget}");
-            Stopwatch stopwatch = Stopwatch.StartNew();
+            using var pageCapacity = new SemaphoreSlim(pageLimit.HasValue ? pageLimit.Value : 100);
 
-            var channel = Channel.CreateBounded<(bool last, ValueTask<FasterKV<Key, Value>.ReadAsyncResult<EffectTracker, Output, object>> responseTask)>(200);
-            using var leftToFill = new SemaphoreSlim(pageLimit.HasValue ? pageLimit.Value : 100);
             using var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(this.partition.ErrorHandler.Token);
             var cancellationToken = cancellationTokenSource.Token;
 
-            Task readIssueLoop = Task.Run(ReadIssueLoop);
-            async Task ReadIssueLoop()
+            this.partition.EventDetailTracer?.TraceEventProcessingDetail($"query {queryId} attempt {attempt:o} enumeration from {queryEvent.ContinuationToken} with pageLimit={(pageLimit.HasValue ? pageLimit.ToString() : "none")} timeBudget={timeBudget}");
+            Stopwatch stopwatch = Stopwatch.StartNew();
```

Comment:
here I'm just re-ordering some variables, to help with readability (it's a long method, I'm trying to group related things together).